### PR TITLE
fix: react-native-windows implementation for new architecture

### PR DESCRIPTION
**Pull request by acoates-ms**
# Summary
There are two main things going on with the PR. Both involve a reworking of the new arch implementation of rn-svg for windows. 
 The current implementation attempts to implement a svg renderer from scratch.  There are numerous edge cases that it wasn't handling correctly, and the performance had some serious issues.  This implementation switches to use the svg rendering path built into Direct2D.  This brings significant performance improvements.

The 2nd issue is there have been various breaking changes in react-native-windows for how new arch native components are implemented.  This brings the rn-svg implementation in line with those latest changes.

## Test Plan

Primary testing right now is loading up the example app in the repo.
New arch on react-native-windows is still in somewhat early days - so there are not really current users of this code.  I am integrating this code into Microsoft Office, where I have tested some scenarios.  But we will get expanded testing as we roll out the new arch.  I expect there to be some follow-ups as we expand our usage.  The version of rn-svg before this PR doesn't build with the latest new arch react-native-windows versions. - So its hard to get worse than that.

## Compatibility

| OS      | Implemented |
| ------- | :---------: |
| iOS     |    N/A  |
| MacOS   |    N/A      |
| Android |    N/A      |
| Web     |    ✅      |

## Checklist

- [x] I have tested this on a device and a simulator
- [ ] I added documentation in `README.md`
- [ ] I updated the typed files (typescript)
- [ ] I added a test for the API in the `__tests__` folder

### CODE REVIEW

File: windows/RNSVG/Fabric/RenderableView.cpp [hunk 1]
```diff
@@ -0,0 +1,174 @@
...
+        return;
+      } else if (strLength > 2) {
+        auto end{strLength - 2};
+        auto lastTwo{str.substr(end)};
+
+        if (lastTwo == "px") {
+          value.units = D2D1_SVG_LENGTH_UNITS::D2D1_SVG_LENGTH_UNITS_NUMBER;
+        } else if (lastTwo == "em") {
+          value.units = D2D1_SVG_LENGTH_UNITS::D2D1_SVG_LENGTH_UNITS_NUMBER;
+          // value.unit = D2D1_SVG_LENGTH_UNITS::D2D1_SVG_LENGTH_UNITS_EM;
```
Comment (marlenecota) [inline windows/RNSVG/Fabric/RenderableView.cpp:39]: did you mean to leave these in?

File: windows/RNSVG/Fabric/SvgView.h [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+
+  // ComponentView
+  void UpdateProps(
+      const winrt::Microsoft::ReactNative::ComponentView & /*view*/,
+      const winrt::Microsoft::ReactNative::IComponentProps &newProps,
+      const winrt::Microsoft::ReactNative::IComponentProps & /*oldProps*/) noexcept;
+  void UpdateLayoutMetrics(
+      const winrt::Microsoft::ReactNative::LayoutMetrics &metrics,
+      const winrt::Microsoft::ReactNative::LayoutMetrics &oldMetrics);
+    void MountChildComponentView(
```
Comment (marlenecota) [inline windows/RNSVG/Fabric/SvgView.h:66]: nit: indentation

File: windows/RNSVG/Fabric/SvgView.h [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+  void Draw(
+      const winrt::Microsoft::ReactNative::Composition::ViewComponentView &view,
+      ID2D1DeviceContext &context,
+      winrt::Windows::Foundation::Size const &size) noexcept;
+
+  bool m_isMounted{false};
+  winrt::Microsoft::ReactNative::Composition::Experimental::ISpriteVisual m_visual{nullptr};
+  winrt::Microsoft::ReactNative::LayoutMetrics m_layoutMetrics{{0, 0, 0, 0}, 1.0};
+  winrt::Microsoft::ReactNative::Composition::Experimental::ICompositionContext m_compContext{nullptr};
+  winrt::weak_ref<winrt::Microsoft::ReactNative::Composition::ViewComponentView> m_wkView; // Do we have the view passed into all the methods that we need this?
```
Comment (marlenecota) [inline windows/RNSVG/Fabric/SvgView.h:98]: is the comment needed?

File: windows/RNSVG/ReactPackageProvider.cpp [hunk 4]
```diff
@@ -60,24 +55,24 @@ namespace winrt::RNSVG::implementation
...
-    TSpanView::RegisterComponent(fabricPackageBuilder);
+    RegisterRectComponent(fabricPackageBuilder);
+    RegisterCircleComponent(fabricPackageBuilder);
+    RegisterEllipseComponent(fabricPackageBuilder);
+    RegisterLineComponent(fabricPackageBuilder);
+    RegisterPathComponent(fabricPackageBuilder);
+    RegisterImageComponent(fabricPackageBuilder);
+    RegisterUseComponent(fabricPackageBuilder);
+    RegisterGroupComponent(fabricPackageBuilder);
+    RegisterUnsupportedSvgComponent(L"RNSVGSymbol", fabricPackageBuilder);
```
Comment (marlenecota) [inline windows/RNSVG/ReactPackageProvider.cpp:66]: Can also add ForeignObject and the Filters to unsupported components.

File: windows/RNSVG/Fabric/RenderableView.cpp (hunk 1 above)
Comment (acoates-ms) [reply windows/RNSVG/Fabric/RenderableView.cpp:41]: Added a comment at the top of the block.

**Review by marlenecota — APPROVED** [review]

**Review by jakex7 — APPROVED** [review]
Thank you @acoates-ms for contribution 🎉 I just left a small suggestion

**Comment by jakex7** [comment]
Hey @acoates-ms, it looks like the CI for `fabric-windows-example` is failing. Could you look into it?

File: windows/RNSVG/ReactPackageProvider.cpp [hunk 5]
```diff
@@ -60,24 +55,26 @@ namespace winrt::RNSVG::implementation
...
-    MaskView::RegisterComponent(fabricPackageBuilder);
-    LinearGradientView::RegisterComponent(fabricPackageBuilder);
-    RadialGradientView::RegisterComponent(fabricPackageBuilder);
-    PatternView::RegisterComponent(fabricPackageBuilder);
-    TextView::RegisterComponent(fabricPackageBuilder);
-    TSpanView::RegisterComponent(fabricPackageBuilder);
+    RegisterRectComponent(fabricPackageBuilder);
+    RegisterCircleComponent(fabricPackageBuilder);
+    RegisterEllipseComponent(fabricPackageBuilder);
+    RegisterUnsupportedSvgComponent(L"RNSVGFilter", fabricPackageBuilder);
```
[the remark this answers is not in the document]
Comment (acoates-ms) [reply windows/RNSVG/ReactPackageProvider.cpp:61]: Done

**Comment by acoates-ms** [comment]
> Hey @acoates-ms, it looks like the CI for `fabric-windows-example` is failing. Could you look into it? (And don't worry about failing E2E 😅)

Oops, I accidently left an Office specific change in the PR, which broke the CI.  Should be fixed now.  I also brought in latest fixes from react-native-windows to deal with some issues we were having with the new arch and cloning custom props objects.

**Comment by jakex7** [comment]
Thank you!

**Pull request merged**